MAUI/DataForm/DataForm-settings.md

```diff
@@ -0,0 +1,853 @@
+{
+    if (e.DataFormItem != null)
+    {
+        if (e.DataFormItem.FieldName == "MiddleName")
+        {
+             e.DataFormItem.IsReadOnly = true;
+        }
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+#### Change the autogenerated editor visibility
+
+You can change the specific editor visibility by using the `IsVisible` property in the `DataFormItem`.
+
+Here, `LastName` data field will be hidden.
+
+{% tabs %}
+{% highlight C# %}
+
+this.dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+{
+    if (e.DataFormItem != null)
+    {
+        if (e.DataFormItem.FieldName == "LastName")
+        {
+             e.DataFormItem.IsVisible = false;
+        }
+    }
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+
+## Setting watermark for auto-generated editor
+
+You can display the watermark in the editor by defining the display attribute or using the `GenerateDataFormItem` event when `AutoGenerateItems` is true.
+
+#### Using attribute
+
+You can show the watermark in the editor by setting the `Prompt` in display attribute.
+
+{% tabs %}
+{% highlight C# %}
+
+[Display(Prompt = "Enter middle name")]
+public string MiddleName { get; set; }
+    
+{% endhighlight %}
+{% endtabs %}
+
+#### Using event
+
+You can show the watermark in the editor by using the `PlaceholderText` property in `DataFormItem`.
+
+{% tabs %}
+{% highlight C# %}
+
+this.dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+{
+    if (e.DataFormItem != null)
+    {
+        if (e.DataFormItem.FieldName == "Description")
+        {
+            e.DataFormItem.PlaceholderText = "Enter description";
+        }
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+
+#### Changing watermark color for auto-generated editor
+
+You can display the color for the watermark in the editor by using `GenerateDataFormItem` event when `AutoGenerateItems` is true.
+
+You can change the color for the watermark in the editor by using the `PlaceholderColor` property in `DataFormItem`.
+
+{% tabs %}
+{% highlight C# %}
+
+this.dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+{
+    if (e.DataFormItem != null)
+    {
+        if (e.DataFormItem.FieldName == "Description")
+        {
+            e.DataFormItem.PlaceholderText = "Enter description";
+            e.DataFormItem.PlaceholderColor = Colors.MistyRose;
+        }
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+
+## Change padding for auto-generated editor
+
+You can create a space around a label and editor by using `Padding` property in the DataFormItem by using `GenerateDataFormItem` event when `AutoGenerateItems` is true.
+
+Here, `FirstName` data field will be changed from the default position.
```

Comment:
Here, the `FirstName` data field will be changed from the default position.

Reply:
updated